keycloak/openid_client.go

```diff
@@ -4,15 +4,41 @@ import (
 	"fmt"
 )
 
+type openidClientSecret struct {
+	Type  string `json:"type"`
+	Value string `json:"value"`
+}
+
 type OpenidClient struct {
-	Id       string `json:"id,omitempty"`
-	ClientId string `json:"clientId"`
-	RealmId  string `json:"-"`
-	Protocol string `json:"protocol"`
+	Id                      string `json:"id,omitempty"`
+	ClientId                string `json:"clientId"`
+	RealmId                 string `json:"-"`
+	Name                    string `json:"name"`
+	Protocol                string `json:"protocol"`                // always openid-connect for this resource
```

Comment:
I don't have an actionable suggestion, but I do wish Go had some support for immutable fields on a struct for things like `protocol` here.